### PR TITLE
multus: revise kube config/client loading

### PR DESCRIPTION
**Pull request by BlaineEXE**
The previous client loader routine assumed KUBECONFIG would be set in CLI environments. Instead, now take this approach:

1. If KUBECONFIG is set, that is the de-facto override that informs the tool it is being run in a CLI environment.
2. Otherwise, try creating a client from the default kube config file.
3. If that fails, assume the tool is running in a Kubernetes Pod.

This will continue supporting dev/test environments so that building the rook container is not necessary for development. It also allows support for highly opinionated environments (like deployed by OpenShift CSVs) where it's not possible to install the Rook operator without also deploying a CephCluster.

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: cmd/rook/rook/rook.go [hunk 1]
```diff
@@ -217,30 +217,54 @@ func GetOperatorBaseImageCephVersion(context *clusterd.Context) (string, error)
...
+		clientset, err = kubernetes.NewForConfig(restConfig)
+		if err == nil {
+			logger.Infof("created kube client interface from default CLI parameters")
+			return clientset
+		}
+		logger.Debugf("failed to create kube client interface from default CLI parameters; " +
+			"check the default kube config file location to ensure the proper config is present there")
+	}
+
+	logger.Infof("attempting to create kube client interface assuming the application is running in a Kubernetes Pod; " +
```
Comment (travisn) [inline cmd/rook/rook/rook.go:258]: This is the common case for running the operator, right? If so, how about debug level?
```suggestion
	logger.Debug("attempting to create kube client interface assuming the application is running in a Kubernetes Pod; " +
```

**Review by travisn — CHANGES_REQUESTED** [review]
just a small suggestion

File: cmd/rook/rook/rook.go (hunk 1 above)
Comment (BlaineEXE) [reply cmd/rook/rook/rook.go:258]: This is the uncommon case if someone installs using CSV --> they must run from the CLI, and the longer text below is critical to fix problems that have been reported by users.

Comment (travisn) [reply cmd/rook/rook/rook.go:258]: Ah, this is not the path for running in the pod? Makes sense then.

**Review by travisn — APPROVED** [review]

**Pull request merged**